### PR TITLE
Date of birth functions

### PR DESCRIPTION
**Pull request by pipliggins**
Adds a function to find a starting year, starting month, and a transformation for dates provided in 2 digit year format (relevant for South Africa) - the Time module translates anything < 68 to 20xx rather than 19xx (obviously an issue for older subjects). Transformation shifts the epoch point to 22.

### CODE REVIEW

File: adtl/transformations.py [hunk 1]
```diff
@@ -171,3 +180,102 @@ def makeDateTime(date, time_24hr, date_format, timezone):
...
+    if date in [None, ""]:
+        return None
+
+    try:
+        cd = datetime.strptime(date, format)
+    except ValueError:
+        logging.error(f"Could not convert date {date!r} from date format {format!r}")
+        return None
+
+    if cd.year >= 2022 and "y" in format:
```
Comment (abhidg) [inline adtl/transformations.py:275]: I'd make this a non-optional parameter `epoch`. Best to be explicitly specified as 2022 within parsers.

File: tests/test_transformations.py [hunk 2]
```diff
@@ -194,3 +198,104 @@ def test_textIfNotNull(field, return_text, expected):
...
+        ("2023-07-28", "day", 2022, "%Y-%m-%d", 28),
+        ("28/07/2023", "year", 2022, "%Y-%m-%d", None),
+    ],
+)
+def test_splitDate(date, option, epoch, date_format, expected):
+    assert transform.splitDate(date, option, epoch, date_format) == expected
+
+
+@pytest.mark.parametrize(
+    "date,duration,epoch,format,type,expected",
```
Comment (abhidg) [inline tests/test_transformations.py:220]: duration,date

**Review by abhidg — APPROVED** [review]
a typo, otherwise looks good!

File: tests/test_transformations.py (hunk 2 above)
Comment (pipliggins) [reply tests/test_transformations.py]: fixed, thanks!

**Pull request merged**